golang/x/relay/keeper/requests.go

```diff
@@ -127,16 +154,16 @@ func (k Keeper) checkRequests(ctx sdk.Context, inputIndex, outputIndex uint8, vi
 			return false, types.ErrRequestPays(types.DefaultCodespace)
 		}
 		paysValue := req.PaysValue
-		if paysValue != 0 || uint64(btcspv.ExtractValue(out)) < paysValue {
+		if paysValue != 0 && uint64(btcspv.ExtractValue(out)) < paysValue {
 			return false, types.ErrRequestValue(types.DefaultCodespace)
 		}
 	}
 
-	hasSpends := req.Spends != types.Hash256Digest{}
+	hasSpends := req.Spends != btcspv.Hash256([]byte{0})
 	if hasSpends {
 		in := btcspv.ExtractInputAtIndex(vin, inputIndex)
```

Comment:
think this is a bug. should extract the outpoint and hash that

Reply:
If that's the case, you should check the solidity instance of this because it uses `ExtractInputAtIndex` as well.

Reply:
Also, that would mean that we don't need an `InputIndex`, correct?

Reply:
Or wait, I would extract the input and then extract the outpoint?

Reply:
Ok, I looked back at the Solidity example.  It does extract the outpoint.  It just does it inline so I must have missed it.